collections-lists/FastestWayToCheckIfListIsOrdered/FastestWayToCheckIfListIsOrdered/Benchmarks.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Buffers;
+using BenchmarkDotNet.Attributes;
+
+namespace FastestWayToCheckIfListIsOrdered;
+
+[MemoryDiagnoser]
+[MarkdownExporterAttribute.Default]
+public class Benchmarks
+{
+    
+    [Params(100, 10_000, 1_000_000)]
+    public int Length { get; }
+
+    private ArrayPool<int> _pool;
+
+    public int[] Array;
+
+    public List<int> List;
+
+    [GlobalSetup]
+    public void Setup()
+    {
+        _pool = ArrayPool<int>.Shared;
+        Array = _pool.Rent(Length);
+        
+        for (var i = 0; i < Length; i++)
+        {
+            Array[i] = i;
+        }
+        
+        List = Array.ToList();
+    }
```

Comment:
There isn't really a need for the ArrayPool here (although kudos for thinking about using it). In a benchmark the GlobalSetup will only be run once, so it does not affect the benchmarking results. In this context, clarity is to be favored over performance. We should just use a line like:
`List = Enumerable.Range(0, Length).ToList()`
Then we can also drop the GlobalCleanup.